Mathlib/LinearAlgebra/Matrix/LinearProgramming.lean

```diff
@@ -0,0 +1,84 @@
+/-
+Copyright (c) 2023 Martin Dvorak. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Martin Dvorak
+-/
+import Mathlib.LinearAlgebra.Matrix.ToLin
+import Mathlib.LinearAlgebra.AffineSpace.AffineMap
+
+/-! # Linear programming
+
+TODO
+
+## Main definitions
+
+ * `CanonicalLP` defines a linear program in the canonical form.
+ * `CanonicalLP.Admits` tells if given vector is an admissible solution to given canonical LP.
+ * `StandardLP` defines a linear program in the standard form.
+ * `StandardLP.Permits` tells if given vector is an admissible solution to given standard LP.
+
+-/
+
+section canonical
+
+open Matrix
+
+/-- Linear program in the canonical form. -/
+structure CanonicalLP (m n K : Type _) [Fintype m] [Fintype n] [LinearOrderedField K] where
+  /-- (possibly not a) matrix of coefficients -/
+  A : (n → K) →ₗ[K] (m → K)
+  /-- Right-hand side -/
+  b : m → K
+  /-- Objective function coefficients -/
+  c : n → K
```

Comment:
Martin, this is a worst-of-both-worlds choice that is halfway between your proposed matrix-only version, and my vector-space-only version. I suggest reverting to your initial version.